files/synchronize.py

```diff
@@ -173,6 +173,8 @@
    - Expect that dest=~/x will be ~<remote_user>/x even if using sudo.
    - To exclude files and directories from being synchronized, you may add 
      C(.rsync-filter) files to the source directory.
+   - Does not use SSH agent, authentication must be done using keys using a
```

Comment:
Could you elaborate about the ssh agent? `PasswordAuthentication=no` doesn't prevent the use of the agent.
In any case, ansible should be able to work with an ssh agent. Removing key files passphrases is a security risk for a lot of users.